### DISCUSSION TITLE
Oauth access token found in commit

### DISCUSSION THREAD

**Discussion by AlexOkoro101** [community/community#24751] [Apps, API and Webhooks]
<p>I’m trying to use Github GraphQL API on my javascript App, but when i pass the token in post header, commit the change and push to remote. The access token gets revoked. How can i fix this? Thank you</p>

**Comment by parithiban** [comment, marked as answer]
<p><a class="mention" href="/alexokoro101">@AlexOkoro101</a></p>
<p>Welcome to GitHub Community</p>
<p>You should not commit your Access token in your repo it will get revoked. Have a look at the <a href="https://github.blog/2015-02-05-keeping-github-oauth-tokens-safe/" rel="noopener nofollow ugc">blog</a> page. Since you are using Javascript application you can set your access token as an environment variable and then access that variable in your application. This would be more helpful as you can have different set of tokens for different environments like (dev,staging,prod)</p>
